io.openems.edge.core/src/io/openems/edge/core/meta/MetaImpl.java

```diff
@@ -36,18 +36,21 @@ public MetaImpl() {
 	}
 
 	@Activate
-	private void activate(ComponentContext context) {
+	private void activate(ComponentContext context, Config config) {
 		super.activate(context, SINGLETON_COMPONENT_ID, Meta.SINGLETON_SERVICE_PID, true);
 
+		this.channel(Meta.ChannelId.CURRENCY).setNextValue(config.currency());
```

Comment:
We often use a `_setCurrency()` helper method in other Interfaces. Might be an option here.

Also usually when I have to apply a config in Activate and Modified, I create a `applyConfig()` method.